src/main/flight/imu.c

```diff
@@ -495,14 +493,15 @@ static void imuCalculateEstimatedAttitude(timeUs_t currentTimeUs)
     if (!useMag && sensors(SENSOR_GPS) && STATE(GPS_FIX) && gpsSol.numSat > GPS_MIN_SAT_COUNT && gpsSol.groundSpeed >= GPS_COG_MIN_GROUNDSPEED) {
         // Use GPS course over ground to correct attitude.values.yaw
         courseOverGround = DECIDEGREES_TO_RADIANS(gpsSol.groundCourse);
-        useCOG = true;
-
+        if (FLIGHT_MODE(GPS_RESCUE_MODE)) {
+            cogYawGain = gpsRescueGetImuYawCogGain(); // do not modify IMU yaw gain unless in a rescue
+        }
```

Comment:
```suggestion
        } else {
            cogYawGain = 1.0f;
        }
```

CoG is used to correct yaw in forward flight. Ideally, thrust vector horizontal vector shall be computed from roll/pitch and used to adjust `courseOverGround`. But for now, don't change existing behavior.

Reply:
Here I think the alternative,or `else` condition, on line 498, should be a zero value, not a 1, to return the float equivalent of the boolean OFF.  That's why I initialise `cogYawGain` to `0.0f` on line 477.  Our intent is to do nothing unless we are in `FLIGHT_MODE(GPS_RESCUE_MODE)`.  If we keep `cogYawGain` at zero, it won't pass the check for `if (cogYawGain != 0.0f)` back on line 213.

I agree it would be great to get a vector from roll/pitch attitude to help evaluate courseOverGround, especially if integrated over time, and also agree that for now I'd not want to change the existing behaviour.

Reply:
@ledvinap Looks like this was merged before I had time to resolve this.

I'll make a small PR to restore the original behaviour.  I'm not 100% exactly sure what is best, but I guess we should correct the quad's attitude throughout the flight, since otherwise it could drift.  And in any case I had no intention of altering the existing behaviour.

Also I was reflecting on your comment, earlier, about using Pitch angle as a factor for IMU gain.

This could be done quite easily in the GPS Rescue case, and would be beneficial in the situation of a rescue initiated with very low groundspeed but an IMU that is 180 degrees wrong.  Currently the quad will pitch forward, but it takes time to accumulate velocity before the IMU gain for yaw is increased.  If we had a signal to boost IMU directly related to forward pitch angle (before adjustment for groundspeed error), this is nearly always high to maximal in groundspeed error settings, and is high from the moment the rescue starts.  Using positive pitch angle to generate IMU yaw gain would act as a leading indicator of a 180 degree IMU error.  Also, generally speaking, it is more likely that if the pitch angle is positive, then we should have direction over ground that corresponds 'better than usually' to the orientation of the nose of the quad.

A flight home at high angle against a strong headwind, or with a high target velocity, could require a high pitch angle all the time.  The quad would then fly home with an increased IMU gain on yaw the whole time.  In both these situations we should be pointing fairly directly to home, and the IMU error in any case resolves to reduce the effect of sideways drag.  IMU yaw gain is functionally a time constant, so we are not going to change the magnitude of anything, just how quickly it happens, and this is probably not so bad in higher pitch angle environments.  

I'm inclined to test this out.  It will require some heuristic gain factor though :-)

Thanks again for all your input to date, I very much appreciate everything you've done here.